scripts/pythondistdeps.py

```diff
@@ -11,21 +11,72 @@
 # RPM python dependency generator, using .egg-info/.egg-link/.dist-info data
 #
 
-# Please know:
-# - Notes from an attempted rewrite from pkg_resources to importlib.metadata in
-#   2020 can be found in the message of the commit that added this line.
-
 from __future__ import print_function
 import argparse
-from os.path import basename, dirname, isdir, sep
-from sys import argv, stdin, version
 from distutils.sysconfig import get_python_lib
+from itertools import chain
+from os.path import basename, dirname, isdir, sep
+from packaging.requirements import Requirement
+from packaging.version import parse
+import re
+from sys import argv, stdin
 from warnings import warn
 
+try:
+    from importlib.metadata import PathDistribution
+    from pathlib import Path
+except ImportError:
+    from importlib_metadata import PathDistribution
+    from pathlib2 import Path
```

Comment:
I'd rather have those two split, to use pathlib + importlib_metadata on e.g. Python 3.7.